codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Headers.java

```diff
@@ -175,6 +176,11 @@ protected void validateName(NameValidator<CharSequence> validator, boolean forAd
     protected void validateValue(ValueValidator<CharSequence> validator, CharSequence name, CharSequence value) {
         // This method has a noop override for backward compatibility, see https://github.com/netty/netty/pull/12975
         super.validateValue(validator, name, value);
+        // https://datatracker.ietf.org/doc/html/rfc9113#section-8.3.1
+        if ((value == null || value.length() == 0) && hasPseudoHeaderFormat(name) && getPseudoHeader(name) == PATH) {
+            PlatformDependent.throwException(connectionError(
+                    PROTOCOL_ERROR, "HTTP/2 pseudo-header ':path' must not be empty.", name));
+        }
```

Comment:
These lines should be moved into the `VALUE_VALIDATOR` on line 86.

Reply:
The `VALUE_VALIDATOR` receives only the value. How can we match name `:path` with the value?
https://github.com/netty/netty/blob/4.1/codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Headers.java#L86-L95

Also `Http2FrameCodecBuilder ` always builds `DefaultHttp2HeadersDecoder` with value validation disabled.
https://github.com/netty/netty/blob/4.1/codec-http2/src/main/java/io/netty/handler/codec/http2/Http2FrameCodecBuilder.java#L208-L210

Reply:
Thanks for finding this missed case @violetagg.
After reading the whole [RFC section 8.3](https://datatracker.ietf.org/doc/html/rfc9113#section-8.3), it sounds like any pseudo-header filed must not be empty, not only the `:path`. The only legit way is to omit a pseudo-header. WDYT?

Performing the check inside `DefaultHttp2Headers` seem reasonable, we just need to find a good spot where we have access to both name and value. I don't have any better idea except `validateValue` method as it's right now.

Reply:
Agree

Reply:
> The `VALUE_VALIDATOR` receives only the value.

Oh, right. I missed that.

Then you can guard the check with a predicate on `valueValidator() != ValueValidator.NO_VALIDATION`, so we skip it when validation is disabled.